src/main/java/org/truffleruby/core/regexp/RegexpNodes.java

```diff
@@ -497,10 +501,11 @@ public static RegexpSetLastMatchPrimitiveNode create() {
         public DynamicObject setLastMatchData(VirtualFrame frame, DynamicObject matchData) {
             if (threadLocalNode == null) {
                 CompilerDirectives.transferToInterpreterAndInvalidate();
-                threadLocalNode = insert(FindThreadAndFrameLocalStorageNodeGen.create(LAST_MATCH_VARIABLE));
+                lastMatchSymbol = getContext().getSymbolTable().getSymbol(LAST_MATCH_VARIABLE);
```

Comment:
You could use `@Cached` instead of a manual field and mixing it with the lazy creating of the node here.

Reply:
Using `@Cached` for the `FindThreadAndFrameLocalStorageNode` would be nicer and more readable as well.

Reply:
Those nodes are going to be disappearing anyway.